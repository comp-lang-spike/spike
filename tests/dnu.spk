#!/usr/bin/env spike


class Foo {
    self doesNotUnderstand: aMessage {
        stdout.printf("I don't understand %s with args %s!\n",
                      aMessage.selector.printString,
                      aMessage.arguments.printString);
        return 42;
    }
}


tuple(...args) { return args; }


main(argv) {
    obj f, args;
    
    f = Foo.new();
    f hey: 42;
    stdout.printf("tomato is %d\n", f.tomato);
    f + 42;
    f();
    args = tuple(1,2,3);
    f(...args);
    return 0;
}
