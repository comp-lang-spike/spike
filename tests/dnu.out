I don't understand $hey: with args { 42 }!
I don't understand $tomato with args {}!
tomato is 42
I don't understand $__add__ with args { 42 }!
I don't understand $__apply__ with args {}!
I don't understand $__apply__ with args { 1, 2, 3 }!
